velox/exec/NestedLoopJoinProbe.cpp

```diff
@@ -502,13 +504,11 @@ RowVectorPtr NestedLoopJoinProbe::getNextCrossProductBatch(
     const std::vector<IdentityProjection>& buildProjections) {
   VELOX_CHECK_GT(buildVector->size(), 0);
 
-  // TODO: For now we only enable the build optimizations in cross-joins, but we
-  // should allow it for other join types as well.
-  if (isCrossJoin() && isSingleBuildRow()) {
+  if (supportSingleBuild() && isSingleBuildRow()) {
```

Comment:
Can you make sure the following case is covered in unit test?  If yes paste the unit test names here, if not add the missing ones:
1. Single row build side inner join with filter
2. Single batch build side inner join with filter

Reply:
Single batch already exists, added single row.